internal/transport/controlbuf.go

```diff
@@ -148,7 +149,8 @@ type dataFrame struct {
 	streamID  uint32
 	endStream bool
 	h         []byte
-	d         []byte
+	d         mem.BufferSlice
+	r         *mem.Reader
```

Comment:
One letter variable names in the existing code are not great. Can we name this `reader` instead?

Reply:
Can do!